backend/email-service/src/main/java/com/karankumar/bookproject/book/service/FakeEmailService.java

```diff
@@ -0,0 +1,32 @@
+package com.karankumar.bookproject.book.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+
+@Profile({"dev", "test"})
+@Component
+public class FakeEmailService implements EmailService {
```

Comment:
As discussed, I don't think we need this

Reply:
If we are following the solution to use `@Profile` we need an implementation fake, because on `dev environment` spring will not know any implementation and the application won't startup.

You can try it on.... the dev environment should be the same as the prod, in other words, we have to have the same behavior, even if it's fake....

BTW, what's your solution?

Reply:
What if we were to have `@Profile("!dev")`, instead? This should then only call the real email service on prod (for the test environment, we can go back to mocking).

Reply:
This wouldn't change that much, because when you use `@Profile`, spring needs the implementations of that class, based on the profiles you have(In our case, we have 2 implementations, one for **prod** and another one for **test** and **dev**), if it is not present, when you run the project, it gives the error "No qualifying bean of type found for dependency", because spring uses the dependency to inject on controller.... We can't use `@Profile` with only one implementation for one envie, one implementation should be for all envies or you have to have another implementations

Reply:
That makes sense, thanks